client/asset/dcr/dcr.go

```diff
@@ -584,7 +576,7 @@ func (dcr *ExchangeWallet) Balance() (*asset.Balance, error) {
 	if err != nil {
 		return nil, err
 	}
-	ab, err := dcr.wallet.AccountBalance(dcr.ctx, dcr.acct, 0)
+	ab, err := dcr.wallet.Balance(dcr.ctx, 0)
```

Comment:
I'm struggling to imagine a scenario when the `ExchangeWallet` would need to know the account number because the custom `Wallet` doesn't. As an example, if we ever implement a privacy `Wallet`, will we need to re-write `ExchangeWallet` to be aware of mixed and unmixed account? These are implementation details. Let's leave `ExchangeWallet` out of it. 

> Decred wallets differentiate wallet balance from account balance, so wallet.Balance reads as though it's fetching the wallet balance

That's an implementation detail that the `ExchangeWallet` doesn't need to care about.

> Also, I don't other wallet implementations having to figure out when to use the configured account,

I don't quite follow. What other account would the wallet implementation use besides the configured one, and when would the `ExchangeWallet` know it, but not the `Wallet`?

> Also covers cases where the ExchangeWallet is reconfigured and the account changes

New `ExchangeWallet` is created on reconfiguration anyway.

Reply:
The "account" config option itself is part of the `WalletDefinition`. If RPC wallet wasn't built-in, we wouldn't even be decoding the setting. `ExchangeWallet` shouldn't be relying on it's presence.